### ISSUE TITLE
Extending the selection of a TextBox doesn't scroll correctly

### ISSUE THREAD

**Issue by MrJul** [AvaloniaUI/Avalonia#14903]
### Describe the bug

Try to scroll a `TextBox` by extending its selection using the mouse, and observe the scrollbar resetting to the top:

https://github.com/AvaloniaUI/Avalonia/assets/1623034/8c457ae0-739a-4ef2-b1b3-7c36229637df

### To Reproduce

```xaml
  <TextBox
    AcceptsReturn="True"
    TextWrapping="Wrap"
    Width="200"
    Height="100"
    Text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus magna. Cras in mi at felis aliquet congue. Ut a est eget ligula molestie gravida. Curabitur massa. Donec eleifend, libero at sagittis mollis, tellus est malesuada tellus, at luctus turpis elit sit amet quam. Vivamus pretium ornare est." />
```

### Expected behavior

The TextBox correctly scrolls.

### Avalonia version

11.0.10, master 27c49a8b2200e00efb744b7132c2774edf2ea366

### Additional context

Works fine in 11.0.9, broken in 11.0.10 and master.

**Comment by MrJul** [comment]
Caused by #14627, reverting that PR fixes the issue.

**Comment by Gillibald** [comment]
I guess we have to sync the last character elsewhere then. Going to revert the PR.

**Comment by MrJul** [comment]
Fixed by #14952
